sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -180,6 +181,27 @@
  *         .withBatchElements(100)); // every batch will have 100 elements
  * }</pre>
  *
+ * <p>Configure timeout for writes:
+ *
+ * <pre>{@code
+ * // Let each attempt run for 1 second, retry if the attempt failed.
+ * // Give up after the request is retried for 60 seconds.
+ * Duration attemptTimeout = Duration.millis(1000);
+ * Duration operationTimeout = Duration.millis(60 * 1000);
+ * data.apply("write",
+ *     BigtableIO.write()
+ *         .withProjectId("project")
+ *         .withInstanceId("instance")
+ *         .withTableId("table")
+ *         .withAttemptTimeout(attemptTimeout)
```

Comment:
Just trying to understand the fix. If the timeout is not set, is the pipeline still could stuck happen? When timeout set, is it true that bundle could fail when timeout and get retried?

Reply:
I think closing the service in tearDown() instead of finish_bundle() is the root cause of the hanging issue in a customer ticket, so this fix should significantly lower the chance of that happening. However, the underlying batch also seems to hang occasionally, and closing it with a timeout can mitigate that, which will cause the bundle the fail but dataflow will retry it.